### ISSUE TITLE
In textract-pretty-printer tables to markdown conversion (sometimes) injects wrong table

### ISSUE THREAD

**Issue by dzmitry-kankalovich** [aws-samples/amazon-textract-textractor#291]
I've been testing `textract-pretty-printer` and pushing it to the limits with complex tables. I've noticed something unusual: most of the time it just works, but sometimes completely wrong table gets injected into resulting markdown. It is hard for me to say if this is a problem with `textract` service or `textract-pretty-printer` though.

Here is example:
<img width="2560" alt="Screenshot 2024-01-02 at 16 20 38" src="https://github.com/aws-samples/amazon-textract-textractor/assets/6346981/38feb739-27e8-44d7-bcfa-068ccacce716">

Notice how original PDF document and markdown have completely different tables on the same page. 

The file used is this one: 
[EPAM Fact Sheet Q1 2023.pdf](https://github.com/aws-samples/amazon-textract-textractor/files/13811394/EPAM.Fact.Sheet.Q1.2023.pdf)

The code snippet I used to generate MD:
```python
import logging
import os
import sys
import uuid

import boto3
from textractcaller.t_call import call_textract, Textract_Features
from textractprettyprinter.t_pretty_print import get_text_from_layout_json

logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)

AWS_REGION = 'us-east-1'

s3 = boto3.client('s3', region_name=AWS_REGION)

file_path = 'EPAM Fact Sheet Q1 2023.pdf'
file_name = os.path.basename(file_path)
bucket_name = '<bucket>'
object_key = f'document/{file_name}'

print(f"Using [{file_path}]...")

try:
    s3.head_object(Bucket=bucket_name, Key=object_key)
    print(f"Found {object_key}. Skip uploading")
except s3.exceptions.ClientError as e:
    if e.response['Error']['Code'] == '404':
        print(f"File {object_key} does not exist. Uploading...")
        s3.upload_file(file_path, bucket_name, object_key)
        print("Upload done.")
    else:
        # Handle other exceptions such as 403 Forbidden, 500, etc.
        print("Error occurred: ", e)

s3_link = f"s3://{bucket_name}/{object_key}"
features = [Textract_Features.LAYOUT, Textract_Features.TABLES]
client_request_token = str(uuid.uuid5(
    uuid.NAMESPACE_URL,
    f"{s3_link}#{features}"
))

logging.info(f"client_request_token: {client_request_token}")

POLL_INTERVAL_SECONDS=5

textract_client = boto3.client('textract', region_name=AWS_REGION)
textract_json = call_textract(
    input_document=s3_link,
    client_request_token=client_request_token,
    features=features,
    boto3_textract_client=textract_client,
    force_async_api=True,
    job_done_polling_interval=POLL_INTERVAL_SECONDS
)
layout = get_text_from_layout_json(
    textract_json=textract_json,
    generate_markdown=True,
)

with open(file_path + ".md", "w", encoding="utf-8") as f:
    f.write("".join([
        f"# Page: {page}\n\n{content}" for page, content in layout.items()
    ]))
```

**Comment by anjanvb** [comment]
PR for the fix - #305

**Comment by schadem** [comment]
published as part of 0.1.9
